### ISSUE TITLE
(6.0) Plugins not loaded unless explicitly referenced

### ISSUE THREAD

**Issue by ElteHupkes** [MvvmCross/MvvmCross#2923]
### Steps to reproduce :scroll:

1. Create an MvvmCross project including the `MvvmCross.Plugin.ResourceLoader` package. I experienced this in iOS, yet to test Android, but let's use iOS as an example. Don't use any of the platform specific classes in code explicitly. This is typical usage, you'd retrieve the resource loader by its interface as in step 2:

2. In a ViewModel or ViewController, try to `Mvx.Resolve<IMvxResourceLoader>()`.

### Expected behavior :thinking:
The resolve should return an instance of the `MvvmCross.Plugin.ResourceLoader.Platforms.Ios.MvxIosResourceLoader` class.

### Actual behavior :bug:
The application crashes with a resolve exception or, if you're doing this during initialisation like I did, just freezes without any warnings.

This happens even if the linker behavior is set to "Don't link". As far as I've been able to discern from reading source code, it is because by default setup scans `AppDomain.CurrentDomain.GetAssemblies()` for plugins. This list only contains assemblies that are already loaded, and because assembly loading is lazy this will not be the case unless the platform specific plugin assembly is explicitly referenced anywhere. Just creating a (`Preserve`d) method with the plugin in the signature somewhere fixes the problem, which is also why there previously were no problems when there were still bootstrap files.

I can't presently think of any solution to this problem that doesn't require some manual work of the user installing the plugin, unless bootstrap files are brought back. It should perhaps be documented somewhere in case users wonder why a plugin mysteriously doesn't load.

### Configuration :wrench:

**Version:** 6.0.1

**Platform:** 
- [x] :iphone: iOS
- [ ] :robot: Android
- [ ] :checkered_flag: WPF
- [ ] :earth_americas: UWP
- [ ] :apple: MacOS
- [ ] :tv: tvOS
- [ ] :monkey: Xamarin.Forms

**Comment by Cheesebaron** [comment]
Please try 6.1.0-beta.1. Should be fix there.

**Comment by ElteHupkes** [comment]
@Cheesebaron Great! I searched for an existing issue before posting but couldn't find one. Can you tell me how it was fixed / refer me to the commit that fixed it? I'm interested in the workings of MvvmCross, find it makes my development much easier if I know what's going on.

**Comment by martijn00** [comment]
This is fixed in https://github.com/MvvmCross/MvvmCross/pull/2865

**Comment by ElteHupkes** [comment]
Am I mistaken or is this once again an issue since #2958 basically reverted #2865?

**Comment by Cheesebaron** [comment]
@ElteHupkes your WPF change broke all other platforms. So yes it was reverted.

**Comment by ElteHupkes** [comment]
@Cheesebaron No need to defend it, I was just curious if I saw that correctly. I'm personally doing manual plugin loads now anyway because is shaves half a second of my startup time not having to do that assembly sweep :).

**Comment by UnreachableCode** [comment]
@ElteHupkes any documentation on how to do this manually? I cannot find any.

**Comment by ElteHupkes** [comment]
@UnreachableCode It's been a while and I don't have access to the code I worked on back then anymore, so I can't definitively tell you what I did. However, `MvxSetup` has a [virtual `LoadPlugins` method](https://github.com/MvvmCross/MvvmCross/blob/develop/MvvmCross/Core/MvxSetup.cs#L414) that loads the plugins. It in turn calls another virtual method, [`GetPluginAssemblies`](https://github.com/MvvmCross/MvvmCross/blob/develop/MvvmCross/Core/MvxSetup.cs#L389), which performs an assembly sweep to get the plugin assemblies. If I had to guess I'd say I probably created an override for the latter in my custom `Setup` class to return the assemblies for the plugins that I used.
